app/src/main/java/de/grobox/transportr/trips/detail/TripDrawer.kt

```diff
@@ -165,7 +168,11 @@ internal class TripDrawer(context: Context) : MapDrawer(context) {
         val drawable: Drawable
         if (type == MarkerType.STOP) {
             drawable = ContextCompat.getDrawable(context, R.drawable.ic_marker_trip_stop) ?: throw RuntimeException()
-            drawable.mutate().setColorFilter(backgroundColor, PorterDuff.Mode.SRC_IN)
+            if (Build.VERSION.SDK_INT >= 29) {
+                drawable.mutate().colorFilter = BlendModeColorFilter(backgroundColor, BlendMode.SRC_IN)
+            } else {
+                drawable.mutate().setColorFilter(backgroundColor, PorterDuff.Mode.SRC_IN)
+            }
```

Comment:
We have these a couple of times. I wonder if there's no better way to do these. Maybe `DrawableCompat#setTint()`

Reply:
Nice, I didn't find this one while being sure that it had to exist. Thanks, I'm going to change it.

Reply:
Actually it turns out not being that easy, I've found several complaints about DrawableCompat not working as expected. For now, I would consider sticking to how it was before as this is at least guaranteed to work on all Android versions.

Reply:
Reverted to how it was before.

Reply:
What a shame :(

Then maybe you can factor this if/else into a UiUtils method, so this drawable tinting is happening in one place only we can later change more easily?

Reply:
I would leave it as is for now, the IDE still complains about deprecated `setColorFilter` so it will be quite easy to find the (only four) places where it should be adapted later on.

Sincerely I don't want to change it for now as I fear to break some old Android versions without testing on various API versions. I guess this could be done in another PR in the future.